lib/ioc_container.dart

```diff
@@ -68,6 +73,9 @@ class IocContainer {
   ///so the container can store scope or singletons
   final Map<Type, Object> singletons;
 
+  // ignore: strict_raw_type, avoid_field_initializers_in_const_classes
+  final Map<Type, AsyncLock> locks;
```

Comment:
I noticed you don't use `pkg:meta` as a dependency I would have suggested `@visibleForTesting` in a bunch of places

Reply:
@jogboms I originally did use that, and at some point, I found a version library conflict with other packages between 1.7 and 1.8. At that point, I made a hard decision to remove all dependencies.

Also, exposing some of the internal workings allows you to access the singletons directly, for example, which was requested in an issue.

Reply:
Actually, you are the meta package is a plague when it comes to version conflicts.